types/telegram-web-app/index.d.ts

```diff
@@ -250,6 +256,26 @@ interface WebApp {
      * in response to a user interaction with the Web App interface (e.g. a click inside the Web App or on the main button).
      */
     readTextFromClipboard(callback?: (data: string | null) => void): void;
+    /**
+     * A method that shows a native popup requesting permission for the bot to
+     * send messages to the user.
+     *
+     * @param callback If an optional callback parameter was passed, the
+     * callback function will be called when the popup is closed and the first
+     * argument will be a boolean indicating whether the user granted this
+     * access.
+     */
+    requestWriteAccess(callback?: (success?: boolean) => void): void;
```

Comment:
`success` contains `true` in case of success and `false` if the user has not granted access
```suggestion
    requestWriteAccess(callback?: (success: boolean) => void): void;
```